bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableClusterUtilities.java

```diff
@@ -0,0 +1,222 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigtable.grpc;
+
+import com.google.bigtable.admin.v2.Cluster;
+import com.google.bigtable.admin.v2.ListClustersRequest;
+import com.google.bigtable.admin.v2.ListClustersResponse;
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.grpc.io.ChannelPool;
+import com.google.common.base.Preconditions;
+import com.google.longrunning.GetOperationRequest;
+import com.google.longrunning.Operation;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+/**
+ * This is a utility to that can be used to resize a cluster. This is useful to use 20 minutes
+ * before a large job to increase Cloud Bigtable capacity and 20 minutes after a large batch job to
+ * reduce the size.
+ */
+public class BigtableClusterUtilities implements AutoCloseable {
+  private static Logger logger = new Logger(BigtableClusterUtilities.class);
+  private final BigtableInstanceName instanceName;
+  private final ChannelPool channelPool;
+  private final BigtableInstanceClient client;
+
+  /**
+   * Constructor for the utility.
+   *
+   * @param projectId The projectId string
+   * @param instanceId The instanceId. Could be '-' for all instances.
+   * @throws IOException
+   * @throws GeneralSecurityException
+   */
+  public BigtableClusterUtilities(String projectId, String instanceId)
+      throws IOException, GeneralSecurityException {
+    this(new BigtableOptions.Builder().setProjectId(projectId).setInstanceId(instanceId).build());
+  }
+
+  /**
+   * Constructor for the utility.
+   * @param options that specify projectId, instanceId, credentials and retry options.
+   * @throws GeneralSecurityException
+   * @throws IOException
+   */
+  public BigtableClusterUtilities(final BigtableOptions options)
+      throws IOException, GeneralSecurityException {
+    this.instanceName =
+        Preconditions.checkNotNull(
+            options.getInstanceName(),
+            "ProjectId and instanceId have to be set in the options.  Use '-' for all instanceIds.");
+    channelPool = BigtableSession.createChannelPool(options.getInstanceAdminHost(), options);
+    client = new BigtableInstanceGrpcClient(channelPool);
+  }
+
+  /**
+   * Gets the server node count of the cluster.
+   * @param clusterId
+   * @param zoneId
+   * @return the {@link Cluster#getServeNodes()} of the clusterId.
+   */
+  public int getClusterSize(String clusterId, String zoneId) {
+    Cluster cluster =
+        Preconditions.checkNotNull(
+            getCluster(clusterId, zoneId), "Cluster " + clusterId + " was not found");
```

Comment:
Error message should have zoneId. I think this goes for every message that prints a cluster...